src/Microsoft.Health.Dicom.Core/Features/Store/StoreDatasetValidator.cs

```diff
@@ -81,9 +81,13 @@ public class StoreDatasetValidator : IStoreDatasetValidator
         }
 
         // validate input data elements
-        if (EnableDropMetadata(_dicomRequestContextAccessor.RequestContext.Version) || _enableFullDicomItemValidation)
+        if (_enableFullDicomItemValidation)
```

Comment:
We don't have this FF enabled for anyone. Should we just remove this? We started with this 2 years hoping someone will ask for it. 
We are not even test this case I think.

If it simplifies we should consider removing it.

There are too many if else in this method